powershell/make-util.js

```diff
@@ -169,8 +203,22 @@ var downloadArchive = function (url) {
 
         // write the completed marker
         fs.writeFileSync(marker, '');
-    }
 
-    return targetPath;
+        resolve(targetPath);
+    })
 }
+
+exports.downloadArchiveAsync = downloadArchiveAsync;
+
+var downloadArchive = function (url, fileName) {
+    var result;
+    downloadArchiveAsync(url, fileName).then(t => result = t);
+    deasync.loopWhile(function () { return result == undefined; });
+    return result;
+}
+
+/**
+ * @deprecated This method uses library which is not prefered to use on production
+ */
+
 exports.downloadArchive = downloadArchive;
```

Comment:
Are we need that for now?

Reply:
since this is exported I am not sure is there any user of this method, thus I prefer to keep for a while.

Reply:
AFAIK, we don't have make files in the final build, so it's only for internal usage